web/pdf_document_properties.js

```diff
@@ -29,32 +31,69 @@ class PDFDocumentProperties {
   /**
    * @param {PDFDocumentPropertiesOptions} options
    */
-  constructor(options) {
-    this.overlayName = options.overlayName;
-    this.fields = options.fields;
-    this.container = options.container;
+  constructor({ overlayName, fields, container, closeButton, }) {
+    this.overlayName = overlayName;
+    this.fields = fields;
+    this.container = container;
 
-    this.rawFileSize = 0;
-    this.url = null;
-    this.pdfDocument = null;
+    this._reset();
 
-    // Bind the event listener for the Close button.
-    if (options.closeButton) {
-      options.closeButton.addEventListener('click', this.close.bind(this));
+    if (closeButton) {
+      // Bind the event listener for the Close button.
+      closeButton.addEventListener('click', this.close.bind(this));
     }
-    this._dataAvailableCapability = createPromiseCapability();
-
     OverlayManager.register(this.overlayName, this.container,
                             this.close.bind(this));
   }
 
   /**
    * Open the document properties overlay.
+   *
+   * Note that since the document properties are fetched asynchronously,
+   * the dialog will *only* be updated if it's still open when the data arrives.
```

Comment:
Good point! I understand what you mean now. The comment itself looks clear after reading it again with this in mind, but I would indeed suggest to move it above the first `OverlayManager` check as the comment describes why we need such a check. Other than moving it, the wording seems fine to me.